modules/currency.js

```diff
@@ -5,7 +5,7 @@ import * as utils from 'src/utils';
 import { config } from 'src/config';
 import { hooks } from 'src/hook.js';
 
-const DEFAULT_CURRENCY_RATE_URL = 'https://currency.prebid.org/latest.json';
+const DEFAULT_CURRENCY_RATE_URL = 'https://cdn.jsdelivr.net/gh/prebid/currency-file@1/latest.json';
```

Comment:
How are we updating this file?  In `npm` it's impossible to republish a given library + version, does jsdelivr not have this restriction, and if not, are we sure they won't have this restriction in the future?

Also, do we have any control over the cache control headers?  Currently jsdelivr seems to be setting it to a week, which isn't very good for a currency conversion file.

Reply:
Source of truth is github, with a new git tag pushed daily via a automated job. We also have access to purge the cache on jsdelivr servers so the cache header shouldn't be an issue. You can see the repo here: https://github.com/prebid/currency-file/releases

Reply:
Good observations @snapwich, but we don't have control over cache control headers with jsdelivr.

Unclear that purging the CDN cache daily will help if the browser's allowed to cache it for a week. I did not do a full analysis of all CDN options, though did ask the community for options to which no one responded.

I will do some quick pricing homework to see if there other options. If not, then 1-week caching is what we get.

> it's impossible to republish a given library + version

We set the version of each new file to 1.SEQUENCE+1, and jsdeliver has the '@1' shortcut to take the latest version of 1.x.

Reply:
Ok - the guy at jsdelivr wants to know what headers we want to set. I think it's just:
```
Cache-Control: max-age=86400
```
@snapwich, @mkendall07 - agree?

Mozilla.org (https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching) suggest that `Cache-Control: max-age` is preferred over `Expires`.

Reply:
`Cache-Control: max-age=86400` looks reasonable to me. Note that this will cache a bit longer than we want - it is relative to the original request - meaning that if you first requested at 10pm than it won't expire until the next day at 10pm. I think it's ok for our use case though.

Reply:
We were using `expires` as an optimization trying to time right after the script updated.  But if we don't have dynamic control over the time then a day is probably okay; worst case scenario someone gets 2 day old data.

Reply:
I've asked jsdelivr if they can add `max-age`.

Also did some pricing from another CDN, which isn't much better than our current one.

Reply:
So apparently they can't add `max-age`. His proposed workaround is to add a cache-buster.

e.g. https://cdn.jsdelivr.net/gh/prebid/currency-file@1/latest.json?date=TODAY

We would modify the module to dynamically replace TODAY with YYYYMMDD, with the effect that each morning the user's browser would see a new query string, causing it to hit the CDN again. Hack, but effective.

So we have a decision:
1) Settle for the one week cache period
2) Implement the daily query string cache buster
3) Keep paying many thousands of dollars per month for hosting this file

Reply:
Can they add `expires` ? 

My vote would be to settle 1 week cache period. This would encourage users to self host if they want more accuracy. The query string would be effective but it's an ugly hack IMO.

Reply:
I don't mind the cache buster, although I've heard mixed results about query string cache busters (vs filename cache busting).

Reply:
He said they're planning on adding cache controls in the future including `Expires`, but did not have a commit date.

Agree that query string is a hack. But it's a hack that preserves existing functionality and saves quite a lot of money, so my vote would be we start off with it as a low-cost workaround to not change the file's loading behavior -- then remove when unnecessary. Am prepared to be outvoted.

Reply:
The promise of future support for cache and/or expires is good enough for me. @bretg can you make the query string change then?

Reply:
Done - @idettman has offered to make a unit test.